File: javascript/rna-transcription/rna-transcription.js
// rna-transcription.js

const dnaTranscriber = function() {
  const dnaToRnaMap = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  }

  transcribe = (nucleotide) => {
    if (nucleotide in dnaToRnaMap) {
      return dnaToRnaMap[nucleotide]
    }
    throw Error("Invalid input")
  }
}

dnaTranscriber.prototype.toRna = function(dna) {
  return [...dna].map(transcribe).join('')
}

module.exports = dnaTranscriber

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    transcription = {
        'G': 'C',
        'C': 'G',
        'T': 'A',
        'A': 'U'
    }

    try:
        return ''.join([transcription[k] for k in dna_strand])
    except:
        raise ValueError('{} is not a valid DNA chain'.format(dna_strand))

